llm_toolkit/models.py

```diff
@@ -263,6 +272,41 @@ class GPT4o(GPT):
   name = 'gpt-4o'
 
 
+class AzureGPT(GPT):
+  """Azure's GPT model."""
+
+  name = 'gpt-3.5-turbo-azure'
+
+  def _get_tiktoken_encoding(self):
+    """Returns the tiktoken encoding for the model."""
+    try:
+      return tiktoken.encoding_for_model(self.name.replace('-azure', ''))
+    except KeyError:
+      logger.info('Could not get a tiktoken encoding for %s.', self.name)
+      return tiktoken.get_encoding('cl100k_base')
+
+  def _get_client(self):
+    """Returns the Azure client."""
+    return openai.AzureOpenAI(azure_endpoint=os.getenv(
+        "AZURE_OPENAI_ENDPOINT", "https://api.openai.com"),
+                              api_key=os.getenv("AZURE_OPENAI_API_KEY",
+                                                "YOUR_AZURE_OPENAI_API_KEY"),
```

Comment:
```suggestion
                              api_key=os.getenv("AZURE_OPENAI_API_KEY"),
```

The other one is not a valid key, and we'd prefer to error if the key is not defined rather than trying with an invalid one.